internal/ingress/controller/kong.go

```diff
@@ -905,9 +953,16 @@ func (n *NGINXController) syncUpstreams(locations []*ingress.Location, backends
 				}
 			}
 
-			err := syncTargets(upstreamName, upstream, client)
-			if err != nil {
-				return errors.Wrap(err, "syncing targets")
+			if kongIngress != nil && kongIngress.Upstream != nil && kongIngress.Upstream.ServiceUpstream {
+				err := syncTargets(upstreamName, upstream, client, true)
+				if err != nil {
+					return errors.Wrap(err, "syncing targets (use service upstream)")
+				}
+			} else {
+				err := syncTargets(upstreamName, upstream, client, false)
+				if err != nil {
+					return errors.Wrap(err, "syncing targets")
+				}
```

Comment:
Could we simplify this logic a little bit?
Probably using a local boolean variable to store whether to use ClusterIP or not and then just use a single call to  `syncTargets`?